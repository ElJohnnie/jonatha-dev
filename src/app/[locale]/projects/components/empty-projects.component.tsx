export default function EmptyBlog({ text }: Readonly<{ text: string }>) {
  return (
    <section className='flex h-full flex-col items-center justify-center'>
      <div className='mx-auto flex text-center lg:max-w-none'>
        <h1 className='mb-4 text-3xl font-extrabold leading-none tracking-tight md:text-6xl dark:text-white'>
          {text}
        </h1>
      </div>
    </section>
  );
}
