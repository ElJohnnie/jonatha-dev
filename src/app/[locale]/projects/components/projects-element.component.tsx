import { ProjectElementComponent } from '../types';
import Link from 'next/link';
import Image from 'next/image';
import { useLocale } from 'next-intl';

export default function ProjectElement({
  project,
}: Readonly<ProjectElementComponent>) {
  const locale = useLocale();
  return (
    <>
      <article
        className='group relative h-64 w-full overflow-hidden rounded-lg bg-cover bg-center shadow-lg transition  duration-300 ease-in-out hover:shadow-2xl'
        style={{
          backgroundImage:
            "url('https://images.unsplash.com/photo-1623479322729-28b25c16b011?ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&ixlib=rb-1.2.1&auto=format&fit=crop&w=1740&q=80')",
        }}
      >
        <div className='absolute inset-0 bg-black bg-opacity-50 transition duration-300 ease-in-out group-hover:opacity-75'></div>
        <div className='relative flex h-full w-full items-center justify-center px-4 sm:px-6 lg:px-4'>
          <h3 className='text-center'>
            <Link
              className='text-center text-2xl font-bold text-white'
              href={`/${locale}/projects/${project.slug}`}
            >
              <span className='absolute inset-0'></span>
              {project.title}
            </Link>
          </h3>
        </div>
      </article>
    </>
  );
}
