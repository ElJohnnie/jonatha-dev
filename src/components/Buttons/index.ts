import AnimateButton from './AnimateButton';

export { AnimateButton };
