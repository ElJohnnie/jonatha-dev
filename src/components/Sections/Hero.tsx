/* eslint-disable react/no-unescaped-entities */
'use client';

import Image from 'next/image';
import AnimateButton from '../Buttons/AnimateButton';

interface HeroProps {
  title: string;
  description?: string;
  solutions?: string[];
  linkButton?: string;
  image?: boolean;
}

const Hero: React.FC<HeroProps> = ({
  title,
  description,
  solutions,
  linkButton,
  image,
}) => {
  return (
    <section className='h-100 mt-12 pt-24'>
      <div className='mx-auto grid max-w-screen-xl justify-center px-4 py-8 lg:grid-cols-12 lg:py-16'>
        <div className='mr-auto place-self-center md:col-span-12'>
          <h1 className='mb-4 max-w-2xl text-6xl font-extrabold leading-none tracking-tight dark:text-white'>
            <span className='leading-12 block w-full bg-gradient-to-r from-indigo-700 to-fuchsia-500 bg-clip-text py-2 text-transparent lg:inline'>
              {title}
            </span>
          </h1>
          {/* <!-- Sliding Text animation --> */}
          {description && (
            <div className='md:text4xl bg-gradient-to-r from-slate-200/60 to-slate-200 to-50% bg-clip-text text-4xl font-extrabold text-transparent [text-wrap:balance]'>
              {description}{' '}
              <span className='inline-flex h-[calc(theme(fontSize.3xl)*theme(lineHeight.tight))] flex-col overflow-hidden text-indigo-500 md:h-[calc(theme(fontSize.4xl)*theme(lineHeight.tight))]'>
                <ul
                  className={`block animate-text-slide-4 text-left leading-tight [&_li]:block`}
                >
                  {solutions &&
                    solutions.map((solution) => (
                      <li key={solution}>{solution}</li>
                    ))}
                  <li aria-hidden='true'>{solutions && solutions[0]}</li>
                </ul>
              </span>
            </div>
          )}
          {/* <!-- End: Sliding Text animation --> */}
          <div className='mt-6'></div>
          {linkButton && (
            <AnimateButton
              title={linkButton}
              route={'/about'}
              icon={
                <svg
                  xmlns='http://www.w3.org/2000/svg'
                  fill='none'
                  viewBox='0 0 24 24'
                  strokeWidth={1.5}
                  stroke='currentColor'
                  className='h-6 w-6'
                >
                  <path
                    strokeLinecap='round'
                    strokeLinejoin='round'
                    d='M17.25 8.25 21 12m0 0-3.75 3.75M21 12H3'
                  />
                </svg>
              }
            />
          )}
        </div>
        {image && (
          <div className='hidden lg:col-span-3 lg:mt-0 lg:flex'>
            <Image
              className='filter-grayscale'
              src={'/images/jonatha.jpg'}
              alt=''
              width={400}
              height={400}
            />
          </div>
        )}
      </div>
    </section>
  );
};

export default Hero;
