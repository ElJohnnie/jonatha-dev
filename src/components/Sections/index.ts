import Hero from './Hero/Hero';
import About from './About/About';
import Blog from './Blog/Blog';

export { Hero, About, Blog };
