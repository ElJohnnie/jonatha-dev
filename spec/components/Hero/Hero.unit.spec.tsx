import { Hero } from '@/components/Sections';
import {  render, } from '@testing-library/react';

jest.mock('next-intl', () => ({
  useTranslations: (key: any) => key,
}));

const heroMockup = {
  title: 'any title',
  description: 'any description',
};

describe('should render correctly Hero component', () => {
  it('match snapshot', () => {
    const {container} = render(<Hero {...heroMockup}></Hero>);
    expect(container.firstChild).toMatchSnapshot();
  });
});
