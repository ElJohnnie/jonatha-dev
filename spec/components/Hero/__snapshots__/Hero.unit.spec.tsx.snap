// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly Hero component match snapshot 1`] = `
<section
  class="h-100 mt-12 pt-24"
>
  <div
    class="mx-auto grid max-w-screen-xl justify-center px-4 py-8 lg:grid-cols-12 lg:py-16"
  >
    <div
      class="mr-auto place-self-center md:col-span-12"
    >
      <h1
        class="mb-4 max-w-2xl text-6xl font-extrabold leading-none tracking-tight dark:text-white"
      >
        <span
          class="leading-12 block w-full bg-gradient-to-r from-indigo-700 to-fuchsia-500 bg-clip-text py-2 text-transparent lg:inline"
        >
          any title
        </span>
      </h1>
      <div
        class="md:text4xl bg-gradient-to-r from-slate-200/60 to-slate-200 to-50% bg-clip-text text-4xl font-extrabold text-transparent [text-wrap:balance]"
      >
        any description
         
        <span
          class="inline-flex h-[calc(theme(fontSize.3xl)*theme(lineHeight.tight))] flex-col overflow-hidden text-indigo-500 md:h-[calc(theme(fontSize.4xl)*theme(lineHeight.tight))]"
        >
          <ul
            class="block animate-text-slide-4 text-left leading-tight [&_li]:block"
          >
            <li
              aria-hidden="true"
            />
          </ul>
        </span>
      </div>
      <div
        class="mt-6"
      />
    </div>
  </div>
</section>
`;
